### ISSUE TITLE
Switch to `SimpleVarInfo` for better computational efficiency

### ISSUE THREAD

**Issue by yebai** [TuringLang/TuringGLM.jl#43]
`SimpleVarInfo` in `DynamicPPL` provides significant speedup over the current implementation. We should consider switching to use it once https://github.com/TuringLang/DynamicPPL.jl/pull/360 is merged.

**Comment by storopoli** [comment]
Does that change how we would write the model inside the `@model` block?

**Comment by yebai** [comment]
It should just work once the `SimpleVarInfo` integration with Turing is complete. But it is perhaps a good idea to test it on GLM models when it becomes available.

**Comment by storopoli** [comment]
@yebai has this been already merged into `Turing.jl`?

**Comment by yebai** [comment]
We are very close now - only one final [PR](https://github.com/TuringLang/DynamicPPL.jl/pull/417) standing. Hopefully, it will get reviewed and merged this week or next.

**Comment by storopoli** [comment]
Ok I subscribed to it and once it is merged and a new Turing version is tagged I will run tests.
